packages/backend/server/src/core/doc-renderer/service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@nestjs/common';
+import { applyUpdate, Doc } from 'yjs';
+
+import { Cache } from '../../fundamentals';
+import { PgWorkspaceDocStorageAdapter } from '../doc';
+import {
+  type PageDocContent,
+  parsePageDoc,
+  parseWorkspaceDoc,
+  type WorkspaceDocContent,
+} from '../utils/blocksuite';
+
+@Injectable()
+export class DocContentService {
+  constructor(
+    private readonly cache: Cache,
+    private readonly workspace: PgWorkspaceDocStorageAdapter
+  ) {}
+
+  async getPageContent(
+    workspaceId: string,
+    guid: string
+  ): Promise<PageDocContent | null> {
+    const cacheKey = `workspace:${workspaceId}:doc:${guid}:content`;
+    const cachedResult = await this.cache.get<PageDocContent>(cacheKey);
+
+    if (cachedResult) {
+      return cachedResult;
+    }
+
+    const docRecord = await this.workspace.getDoc(workspaceId, guid);
+    if (!docRecord) {
+      return null;
+    }
+
+    const doc = new Doc();
+    applyUpdate(doc, docRecord.bin);
+
+    const content = parsePageDoc(doc);
+
+    if (content) {
+      await this.cache.set(cacheKey, content, {
+        ttl:
```

Comment:
ttl seems too long

Reply:
it will be invalid after doc updated

Reply:
The cache should automatically expire when the doc content changes

Reply:
Ah i see. It is in the next PR https://github.com/toeverything/AFFiNE/pull/8088/files#diff-9b038965b8326df2fe79c8ae052c4bebc79d9cd676ca7c8f0c2fe2dc728a86baR81